lua/entities/magicbutton.lua

```diff
@@ -0,0 +1,457 @@
+			TRACE_COUNTER = TRACE_COUNTER + 1
+			trace2 = nil
+		end
+	
+		if(trace1 == nil)then 
+			ORIGIN_FAILURES = ORIGIN_FAILURES + 1 
+			TOTAL_FAILURES = TOTAL_FAILURES + 1 
+			continue
+		else
+			trace2 = self:FindCastFinal(trace1,TRACE1_FAILURES) --only returns a final result if it's considered valid
+		end
+	
+		if(trace2 == nil)then 
+			TRACE1_FAILURES = TRACE1_FAILURES + 1 
+			TOTAL_FAILURES = TOTAL_FAILURES + 1 
+			continue 
+		end
+	end
+	
+	if(trace1 and trace2)then
+		--print("SUCCESS AFTER "..TOTAL_FAILURES.." ATTEMPTS!")
+		return trace2,trace1
+	end
+	--print("SUPER BIG FAILURE! HOW IS IT POSSIBLE?")
+end
+
+function ENT:MoveToTraceResult(trace)
+self:SetPos(trace.HitPos)
+local ang = trace.HitNormal:Angle()
+ang:RotateAroundAxis(ang:Right(),-90)
+ang:RotateAroundAxis(ang:Up(),180)
+self:SetAngles(ang)
+self.HasSpot = true
+end
+
+function ENT:Initialize()
+if(SERVER)then
+	self.Entity:SetModel("models/pyroteknik/secretbutton.mdl");
+	
+	local bmins,bmaxs = Vector(-4.8,-3.1,0),Vector(4.8,3.1,2)
+	self:SetCollisionBounds(bmins,bmaxs)
+	
+	self.Entity:PhysicsInitBox(bmins,bmaxs)
+	self.Entity:SetMoveType(MOVETYPE_NONE);
+	self.Entity:SetCollisionGroup(COLLISION_GROUP_DEBRIS_TRIGGER);
+ 
+	self:SetUseType(SIMPLE_USE)
+	self:SetColor(HSVToColor(math.Rand(0,360),1,1))
+	local phys = self:GetPhysicsObject()
+	if(IsValid(phys))then
+	phys:EnableMotion(false)
+	end
+	--self:FindHidingSpot()
+	if(!self.HasSpot)then
+	local trace = self:FindHidingSpot()
+	self:MoveToTraceResult(trace)
+	end
+end
+if(CLIENT)then
+	local effectdata = EffectData()
+	effectdata:SetOrigin( self:GetPos() )
+	effectdata:SetEntity( self )
+	util.Effect( "propspawn", effectdata )
+	
+	
+end
+end
+
+
+function ENT:SpawnFunction( ply, tr, ClassName )
+
+	
+	local ent = ents.Create( ClassName )
+	ent:SetPos(tr.HitPos)
+	ent:SetAngles(VectorRand():AngleEx(tr.HitNormal))
+	local trace = ply:GetEyeTrace()
+	ent:MoveToTraceResult(trace)
+	ent.HasSpot = true
+	ent:Spawn()
+	ent:Activate()
+	return ent
+end
+
+
+
+
+
+
+function ENT:Think()
+
+end
+
+
+function ENT:OnRemove()
+
+end
+
+function ENT:Draw()
+local pos = self:GetPos() + self:GetUp()*3
+local c = self:GetColor()
+local cvector = (Vector(c.r,c.g,c.b)/255)
+local lc = render.ComputeLighting( pos, self:GetUp() )
+local light1 = {}
+light1.type = MATERIAL_LIGHT_POINT
+light1.color = cvector * 15 
+light1.pos = pos
+light1.fiftyPercentDistance  = 2.35
+light1.zeroPercentDistance  = 4
+render.SuppressEngineLighting( true )
+
+for i=0,5 do
+render.SetModelLighting( i , lc.x,lc.y,lc.z)
+end
+
+render.SetLocalModelLights( {light1} )
+render.SetAmbientLight(0.05,0.05,0.05)
+--render.SuppressEngineLighting( true )
+self:DrawModel()
+render.SuppressEngineLighting( false )
+end
+
+function MoneyPrize() -- this is where we generate a random amount of points to give the player
+	local basenumber = 100
+	for i=1,math.random(10,40) do
```

Comment:
Dude, what???

If you want a non-linear sampling probability (eg, more likely to choose a small number) do something like 

`math.pow(math.random(), 5) * (MAX-MIN) + MIN`

Reply:
i told you i don't know math lol

Reply:
the secret is that math.random() is 0 to 1, and pow(math.random(), anynumber) is also 0 to 1. if anynumber>1, the result biases more toward 0, if anynumber < 1 the result is biased more toward 1

Reply:
for some reason i thought pow() was the same as number^x

Reply:
pow(a,b) is a^b

Reply:
oh right yeah i guess it makes sense that 0-1 always equals 0-1.  usually when i use it i would use values above 1